frontend/src/features/admin-form/settings/components/TwilioSettingsSection/TwilioDetailsInputs.tsx

```diff
@@ -26,48 +38,78 @@ const TWILIO_INPUT_RULES: Record<keyof TwilioCredentials, RegisterOptions> = {
       value: /^AC/,
       message: 'Account SID must start with AC',
     },
+    validate: {
+      noWhitespace: (value) =>
+        !value.trim().match(/\s/) || 'Account SID must not contain whitespace',
+    },
   },
   apiKey: {
     required: 'API key SID is required',
     pattern: {
       value: /^SK/,
       message: 'API key SID must start with SK',
     },
+    validate: {
+      noWhitespace: (value) =>
+        !value.trim().match(/\s/) || 'API key SID must not contain whitespace',
+    },
   },
   apiSecret: {
     required: 'API key secret is required',
+    validate: {
+      noWhitespace: (value) =>
+        !value.trim().match(/\s/) ||
+        'API key secret must not contain whitespace',
+    },
   },
   messagingServiceSid: {
     required: 'Messaging service SID is required',
     pattern: {
       value: /^MG/,
       message: 'Messaging service SID must start with MG',
     },
+    validate: {
+      noWhitespace: (value) =>
+        !value.trim().match(/\s/) ||
+        'Messaging service SID must not contain whitespace',
```

Comment:
/nit: The `trim()` implies we want to allow accidental leading or trailing whitespaces, but the pattern check above disallows leading whitespace 🤔 . We could probably remove the pattern check to do it in the validator instead, or add optional leading whitespaces in the pattern (e.g. `/^\s*MG/`). And same comment for the checks above.

Also nit: all 4 validators are nearly identical pure functions (minus the error message), so we could have a generator in the module for them.

Reply:
inlined whitespace check into pattern validator: 353ba39

Reply:
Thanks!

Although, the regex updates shouldn't check the trailing whitespace, since there is no tail anchor.